core/dbt/events/functions.py

```diff
@@ -1,117 +1,128 @@
-        #     "class": "logging.handlers.WatchedFileHandler",
-        #     "filename": "test.log",
-        #     "formatter": "plain",
-        # },
-    },
-    "loggers": {
-        "": {
-            "handlers": ["default"],  # TODO ["default", "file"],
-            "level": "DEBUG" if flags.DEBUG else "INFO",
-            "propagate": True,
-        },
-    }
-})
+    timestamper = structlog.processors.TimeStamper("%H:%M:%S")
 
-structlog.configure(
-    processors=[
+    pre_chain = [
+        # Add the log level and a timestamp to the event_dict if the log entry
+        # is not from structlog.
         structlog.stdlib.add_log_level,
-        structlog.stdlib.PositionalArgumentsFormatter(),
         timestamper,
-        structlog.processors.StackInfoRenderer(),
-        structlog.processors.format_exc_info,
-        structlog.stdlib.ProcessorFormatter.wrap_for_formatter,
-    ],
-    logger_factory=structlog.stdlib.LoggerFactory(),
-    wrapper_class=structlog.stdlib.BoundLogger,
-    cache_logger_on_first_use=True,
-)
+    ]
+
+    logging.config.dictConfig({
+        "version": 1,
+        "disable_existing_loggers": False,
+        "formatters": {
+            "plain": {
+                "()": structlog.stdlib.ProcessorFormatter,
+                "processor": structlog.dev.ConsoleRenderer(colors=False),
+                "foreign_pre_chain": pre_chain,
+            },
+            "colored": {
+                "()": structlog.stdlib.ProcessorFormatter,
+                "processor": structlog.dev.ConsoleRenderer(colors=True),
+                "foreign_pre_chain": pre_chain,
+            },
+        },
+        "handlers": {
+            "default": {
+                "level": "DEBUG",
+                "class": "logging.StreamHandler",
+                "formatter": "colored",
+            },
+            "file": {
+                "level": "DEBUG",
+                "class": "logging.handlers.WatchedFileHandler",
+                # TODO this default should live somewhere better
+                "filename": os.path.join(logger.LOG_DIR, 'dbt.log'),
+                "formatter": "plain",
+            },
+        },
+        "loggers": {
+            "": {
+                "handlers": ["default", "file"],
+                "level": "DEBUG" if flags.DEBUG else "INFO",
+                "propagate": True,
+            },
+        }
+    })
+
+    structlog.configure(
+        processors=[
+            structlog.stdlib.add_log_level,
+            structlog.stdlib.PositionalArgumentsFormatter(),
+            timestamper,
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.format_exc_info,
+            structlog.stdlib.ProcessorFormatter.wrap_for_formatter,
+        ],
+        logger_factory=structlog.stdlib.LoggerFactory(),
+        wrapper_class=structlog.stdlib.BoundLogger,
+        cache_logger_on_first_use=True,
+    )
 
-log = structlog.get_logger()
+    LOG = structlog.get_logger()
```

Comment:
Why doesn't this get set?

Reply:
This creates a new local instance of LOG and doesn't modify the global instance.

Reply:
```suggestion
    global LOG
    LOG = structlog.get_logger()
```
I think this should do what you expect

Reply:
That did it! Thanks!

Now I am going to read on how Python manages scope because this is not how this works in a lot of other languages.

Reply:
Sounds about right 🐍